src/_png.cpp

```diff
@@ -206,7 +206,17 @@ static PyObject *Py_write_png(PyObject *self, PyObject *args, PyObject *kwds)
             goto exit;
         }
         buff.cursor = 0;
-    } else if ((fp = mpl_PyFile_Dup(py_file, (char *)"wb", &offset))) {
+    } else {
+        #if PY3K
```

Comment:
Just remembered seeing an article recently imploring that we stop writing if-statements like this that would possibly break in py4.x. The logic really should be checking whether or not it is py2.x, as py2.x should be considered the exception to the rule.